src/main/java/org/broadinstitute/hellbender/engine/FeatureDataSource.java

```diff
@@ -396,7 +397,7 @@ final void printCacheStats() {
             final GenomicsDBExportConfiguration.ExportConfiguration exportConfigurationBuilder =
                     createExportConfiguration(workspace, callsetJson, vidmapJson, vcfHeader, genomicsDBOptions);
             return new GenomicsDBFeatureReader<>(exportConfigurationBuilder,
-                    new GenomicsDBBCFCodec(),
+                    new VCFCodec(),
```

Comment:
Wondering about the performance implications of changing to VCF format ? Also, if we no longer need the  `GenomicsDBBCFCodec` codec, that class (and it's tests) can be deleted.

Reply:
Also, there are some other tests where `GenomicsDBBCFCodec` is still used - I'm surprised those don't seem to be failing.

Reply:
GenomicsDBImportIntegrationTest has been ported already to use VCFCodec in the change set for this PR. But, I can parameterize the tests to test both BCFCodec and VCFCodec if decide to keep BCFCodec for performance reasons.

Regarding performance, we will have to benchmark GenomicsDB queries as @droazen mentions. We can probably add a FeatureReader option to choose between BCFCodec and VCFCodec for GenomicsDB if performance turns out to be an issue. If BCFCodec is chosen, GenomicsDB gracefully handles a number of overflow scenarios already.

Note that -
* BCF and BCFCodec do not support 64-bit fields.
* VCFCodec does not fully decode the VCF streams by default, it uses the LazyParser whereas BCFCodec always fully decodes the BCF streams. Hopefully, this keeps the performance somewhat similar even if fields have to be explicitly decoded.
* GenomicsDBImport already uses VCFCodec to read vcfs, so no change there.

Reply:
Ah, ok, I see. So then if we just need to delete the `GenomicsDBBCFCodec` class (unless we keep an option for BCF, in which case it will depend on the version of BCF that uses). The `GenomicsDBBCFCodec` is only necessary if the BCF header is stamped with version 2.2.

Reply:
I did a benchmark run with about 5K samples more than a year back - negligible difference in performance between VCF and BCF serialization/deseralization. It would be good to re-check again with your data.